server/chat-plugins/room-events.js

```diff
@@ -22,15 +22,19 @@ exports.commands = {
 			let buff = '<table border="1" cellspacing="0" cellpadding="3">';
 			buff += '<th>Event Name:</th><th>Event Description:</th><th>Event Date:</th>';
 			for (let i in room.events) {
-				buff += `<tr><td>${Chat.escapeHTML(room.events[i].eventName)}</td><td>${Chat.formatText(room.events[i].desc, true)}</td><td><time>${Chat.escapeHTML(room.events[i].date)}</time></td></tr>`;
+				let timeRemaining = new Date(room.events[i].date).getTime() - new Date().getTime();
+				if (!timeRemaining) timeRemaining = "The time remaining for this event is not available";
+				if (timeRemaining < 0) timeRemaining = "This event will start soon";
+				if (room.events[i].started) timeRemaining = "This event has started";
+				buff += `<tr title="${isNaN(timeRemaining) ? timeRemaining : `This event will start in: ${Chat.toDurationString(timeRemaining, {precision: 2})}`}"><td>${Chat.escapeHTML(room.events[i].eventName)}</td><td>${Chat.formatText(room.events[i].desc, true)}</td><td><time>${Chat.escapeHTML(room.events[i].date)}</time></td></tr>`;
 			}
 			buff += '</table>';
 			return this.sendReply(`|raw|<div class="infobox-limited">${buff}</div>`);
 		},
 		new: 'add',
 		create: 'add',
 		edit: 'add',
-		add(target, room, user) {
+		add(target, room, user, connection, cmd) {
```

Comment:
This is no longer needed

```suggestion
		add(target, room, user) {
```